ddl/partition.go

```diff
@@ -916,6 +916,8 @@ func onDropTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _
 		}
 	}
 
+	// used by ApplyDiff in updateSchemaVersion
+	job.Args = append(job.Args, physicalTableIDs)
```

Comment:
What if `updateVersionAndTableInfo` fails and the dropping partition / rollbacking creating partition process falls into a retry logic? The last `physicalTableIDs` parameters will be taken as partition names.

Reply:
Did not the retry will call `onDropTablePartition` again? Invoking `DecodeArgs` will override `job.Args` by the content of `RawArgs`, that is `physicalTableIDs` will be dropped during `DecodeArgs`.

Also I append the argument, but not directly set.